src/boot/efifirmware.h

```diff
@@ -0,0 +1,28 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include "efi.h"
+
+#define FWHEADERMAGIC (UINT32_C(0xfeeddead))
+
+/*  The structure of the efifw UKI blob is the following:
+ *  ---------------------------------------------------------
+ *  EfiFw header|fwid|payload| reserved for future attributes
+ *  ---------------------------------------------------------
+ *  The base header defines the length of full header, fwid and payload.
+ *  The fwid is a NULL terminated string.
```

Comment:
btw, ASCII (ANSI_X3.4-1968) calls that zero byte "NUL" typically, and C calls the zero pointer NULL usually. not everyone is particular strict or accurate about it, but in our codebase we kinda try to follow that, and would write "NUL terminated string", not "NULL terminated string", and conversely never write "NUL pointer", but only "NULL pointer".

The difference is also mentioned on https://en.wikipedia.org/wiki/Null_character btw